src/main/java/com/shopping_app/shoppingApp/service/CartService.java

```diff
@@ -0,0 +1,97 @@
+package com.shopping_app.shoppingApp.service;
+
+import com.shopping_app.shoppingApp.Exceptions.BaseException;
+import com.shopping_app.shoppingApp.Exceptions.NotFoundException;
+import com.shopping_app.shoppingApp.domain.Cart;
+import com.shopping_app.shoppingApp.domain.CartItems;
+import com.shopping_app.shoppingApp.domain.Product;
+import com.shopping_app.shoppingApp.model.Cart.CartAddRequest;
+import com.shopping_app.shoppingApp.model.Cart.CartItemUpdateRequest;
+import com.shopping_app.shoppingApp.model.Cart.CartItemResponse;
+import com.shopping_app.shoppingApp.model.Cart.CartResponse;
+import com.shopping_app.shoppingApp.repository.CartItemRepository;
+import com.shopping_app.shoppingApp.repository.CartRepository;
+import com.shopping_app.shoppingApp.repository.ProductRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Service
+@Slf4j
+@Transactional
+@RequiredArgsConstructor(onConstructor = @__({@Autowired}))
+public class CartService {
+
+    private final CartRepository cartRepository;
+    private final ProductRepository productRepository;
+    private final CartItemRepository cartItemRepository;
+
+    public CartResponse getCart(Long userId) {
+        Cart userCart = getCartByUserId(userId);
+        Set<CartItemResponse> cartItemResponses = userCart.getCartItems().stream().map(CartItemResponse::from).collect(Collectors.toSet());
+        return CartResponse.from(userCart, cartItemResponses);
+    }
+
+    public CartResponse addItemsToCart(CartAddRequest cartRequest, Long userId) {
+        Cart userCart = getCartByUserId(userId);
+        Optional<CartItems> cartItemProduct = cartItemRepository.findByproductId(cartRequest.getProduct_id());
+        if (cartItemProduct.isPresent()) {
+            throw new BaseException("Product already added in cart", HttpStatus.BAD_REQUEST);
+        }
```

Comment:
What should be the behaviour for an e-commerce site when you try to add a product in the cart which already exists in there?

Reply:
replaced this validation